blueocean-dashboard/src/main/js/credentials/git/GitCredentialsPicker.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { PropTypes } from 'react';
+import { observer } from 'mobx-react';
+import { FormElement } from '@jenkins-cd/design-language';
+import { Fetch, AppConfig } from '@jenkins-cd/blueocean-core-js';
+import { Button } from '../../creation/github/Button';
+import { i18nTranslator } from '@jenkins-cd/blueocean-core-js';
+const t = i18nTranslator('blueocean-dashboard');
+
+function copySelectionText() {
+    let copysuccess; // var to check whether execCommand successfully executed
+    try {
+        copysuccess = document.execCommand('copy'); // copy selected text to clipboard
+    } catch (_) {
+        copysuccess = false;
+    }
+    return copysuccess;
+}
+
+function clearSelection() {
+    if (window.getSelection) {
+        window.getSelection().removeAllRanges();
+    } else if (document.selection) {
+        document.selection.empty();
+    }
+}
+
+@observer
+class GitCredentialsPicker extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {};
```

Comment:
I think at least until we have a solid type system in place we should set initial values for all `state` keys, otherwise it's very hard to understand what the contract is.

Reply:
👍